aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -688,17 +688,36 @@ Tensor select_nested(const Tensor& self, int64_t dim, int64_t index) {
 Tensor clone_nested(
     const Tensor& self,
     c10::optional<c10::MemoryFormat> optional_memory_format) {
-  auto memory_format = optional_memory_format.value_or(MemoryFormat::Preserve);
-  TORCH_CHECK(
-      memory_format == MemoryFormat::Preserve,
-      "clone_nested only supports memory format Preserve, but got ",
-      memory_format,
-      " instead.");
-  // TODO: The size doesn't necessarily need to be cloned, but it is more
-  // conservative. This is something we could revisit once we land a more
-  // efficient implementation of nested_size_tensor_.
-  return wrap_buffer(
-      get_buffer(self).clone(), get_nested_size_tensor(self).clone());
+  auto memory_format = optional_memory_format.value_or(c10::MemoryFormat::Preserve);
+  auto self_ptr = get_nested_tensor_impl(self);
+  if (memory_format == c10::MemoryFormat::Preserve ||
+  (memory_format == c10::MemoryFormat::Contiguous && nested_tensor_impl_is_contiguous(self_ptr))) {
+    const Tensor& buffer = self_ptr->get_buffer(),
+        sizemat = self_ptr->get_nested_size_tensor(),
+        stridemat = self_ptr->get_nested_stride_tensor();
+    const std::vector<int64_t>& offsets = self_ptr->get_offsets();
+    // TODO: The size and the stride do not necessarily need to be cloned,
+    //       but it is more conservative.
+    //       This is something we could revisit once we land a more
+    //       efficient implementation of nested_size_tensor_ and nested_stride_tensor.
+    return wrap_buffer(buffer.clone(), sizemat.clone(), stridemat.clone(), offsets);
+  }
+  // actually, memory format is contiguous and self is noncontiguous
```

Comment:
This is essentially the impl of self.contiguous(), might be cleaner to just override contiguous() for nt and calling  self.contiguous() here, I can just note that for clean up though

Reply:
I found `contiguous` a CompositeImplicit op, which may redispatch to `clone(Contiguous)`, so it seems to me that we should implement `clone(Contiguous)` then leave `contiguous` the same.

Anything I missed?

Reply:
ohh cool thats perfect. So this change should allow us to call `.contiguous` on nested tensors? Maybe we should add a test for this

Reply:
Just tried. Yes we can call `nested_tensor.contiguous()` now